src/Options/Options.hpp

```diff
@@ -0,0 +1,557 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defined classes and functions that handle parsing of input parameters.
+
+#pragma once
+
+#include <exception>
+#include <iterator>
+#include <map>
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+#include <yaml-cpp/yaml.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Options/OptionsDetails.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/PrettyType.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+/// \ingroup OptionParsing
+/// Information about the nested operations being performed by the
+/// parser, for use in printing errors.  A default-constructed
+/// OptionContext is printed as an empty string.  This struct is
+/// primarily used as an argument to PARSE_ERROR for reporting input
+/// file parsing errors.
```

Comment:
Could you add a brief explanation of what `YAML::Mark` is used for or link to the yaml-cpp documentation? The yaml-cpp documentation is not exactly super complete or easy to navigate. If I understand correctly, `YAML::Mark` used to store the line and column in the file where a parsing error occurred?